tests/unit_tests/sid/test_sid_rename_dimensions.cpp

```diff
@@ -38,7 +38,7 @@ namespace gridtools {
                            .set<property::strides>(tu::make<hymap::keys<a, b, c>::values>(5_c * 7_c, 7_c, 1_c))
                            .set<property::upper_bounds>(tu::make<hymap::keys<a, b>::values>(3, 5));
 
-            auto testee = sid::rename_dimension<b, d>(src);
+            auto testee = sid::rename_dimensions(src, b(), d());
```

Comment:
I think we should go back to the old API, because now the keys need to be defined (before declaration was enough).

Reply:
We can still do the new API in the different style. It will be more verbose thought. smth. like
`sid::rename_dimensions<meta::list<meta::list<b, c>>>(src)` and
`sid::rename_numbered_dimensions<meta::list<a, b, c>>(src)`;

Looks ugly on my taste. That is my I leaned to sacrify the ability not to define the dimension tags.
Frankly I don't know the advantages of not having definitions.

Reply:
by providing an extra overload, we can have both, no?
```
            template <class Sid, class... Keys>
            renamed_sid<typename make_key_map<Keys...>::type, Sid> rename_dimensions(Sid &&sid) {
                return {std::forward<Sid>(sid)};
            }
```

In my prototype for the unstructured I have some keys as part of the mesh concept, e.g. `cell`, `edge`, `vertex`. So far I needed only a declaration and I don't have to take care of ODR. It feels cleaner that I don't have to provide that empty struct, but I don't have an example why this is an advantage.

Reply:
another workaround is to use pointers as a parameters. smth. like
 `sid::rename_dimensions(src, (b*){}, (d*){})`

Reply:
What's the advantage of the new API? The options are
- `rename<key,key>(some_other_thing)`
- `rename(some_other_thing, key,key)` .
For me the first looks cleaner because it separates arguments of different semantic clearly.

Reply:
or we can pass key parameters as const refs.
in this case we can use simplified syntax if the dimension tag is defined:
```
struct a{};
struct b;
sid::rename_dimensions(src, a(), *(b*){});
```

Reply:
there is the serious problem with your overload -- it ruins perfect forwarding. you have to explicitly specify `Sid` template parameter if you have any keys.

Reply:
if you like angle brackets so much there is one more alternative -- split the api into three public functions:
`sid::rename_dimensions(src, sid::make_dimension_map<x, i, y, j>());`
`sid::rename_dimensions(src, sid::make_dimension_map_numbered<i, j>());`

Reply:
@havogt PTAL
I think now I've managed to craft an API that you should like.